src/store/messagesStore.spec.js

```diff
@@ -1778,4 +1787,182 @@ describe('messagesStore', () => {
+
+			store.dispatch('addConversation', conversations[1])
+
+			// Act
+			store.dispatch('forwardMessage', { messageToBeForwarded })
+
+			// Assert
+			expect(postNewMessage).toHaveBeenCalledWith(messageExpected, { silent: true })
+		})
+
+		test('generates Note to self when it does not exist ', async () => {
+			// Arrange
+			messageToBeForwarded = message1
+			messageExpected = cloneDeep(message1)
+			messageExpected.token = 'token-self'
+
+			const response = {
+				data: {
+					ocs: {
+						data: conversations[1],
+					},
+				},
+			}
+			fetchNoteToSelfConversation.mockResolvedValueOnce(response)
+
+			// Act
+			store.dispatch('forwardMessage', { messageToBeForwarded })
+			await flushPromises()
+
+			// Assert
+			expect(store.getters.conversationsList).toContain(conversations[1])
+			expect(postNewMessage).toHaveBeenCalledWith(messageExpected, { silent: true })
+		})
+		test('removes parent message ', () => {
+			// Arrange : prepare the expected message to be forwarded
+			messageToBeForwarded = {
+				id: 1,
+				token: TOKEN,
+				parent: message1,
+				message: 'simple text message',
+				messageParameters: {},
+			}
+			messageExpected = cloneDeep(messageToBeForwarded)
+			targetToken = 'token-2'
+			messageExpected.token = targetToken
+			delete messageExpected.parent
+
+			// Act
+			store.dispatch('forwardMessage', { targetToken, messageToBeForwarded })
+
+			// Assert
+			expect(postNewMessage).toHaveBeenCalledWith(messageExpected, { silent: true })
+		})
+		test('forwards an object message', () => {
+			// Arrange
+			messageToBeForwarded = {
+				id: 1,
+				token: TOKEN,
+				message: '{object}',
+				messageParameters: {
+					object: {
+						id: '100',
+						type: 'deck-card',
+					},
+				},
+			}
+			const objectToBeForwarded = messageToBeForwarded.messageParameters.object
+			targetToken = 'token-2'
+
+			// Act
+			store.dispatch('forwardMessage', { targetToken, messageToBeForwarded })
+
+			// Assert
+			expect(postRichObjectToConversation).toHaveBeenCalledWith(
+				targetToken,
+				{
+					objectId: objectToBeForwarded.id,
+					objectType: objectToBeForwarded.type,
+					metaData: JSON.stringify(objectToBeForwarded),
+					referenceId: '',
+				},
+			)
+
+		})
+		test('forwards a message with mentions and remove the latter', () => {
+			messageToBeForwarded = {
+				id: 1,
+				token: TOKEN,
+				message: 'Hello {mention-user1}, {mention-user2}',
+				messageParameters: {
+					'mention-user1': {
+						id: 'taylor',
+						name: 'Taylor',
+						type: 'user',
+					},
+					'mention-user2': {
+						id: 'adam',
+						name: 'Adam',
+						type: 'user',
+					},
+				},
+			}
+			targetToken = 'token-2'
+			const mentions = [
+				messageToBeForwarded.messageParameters['mention-user1'],
+				messageToBeForwarded.messageParameters['mention-user2'],
+			]
+			messageExpected = cloneDeep(messageToBeForwarded)
+			messageExpected.message = `Hello @'${mentions[0].name}', @'${mentions[1].name}'`
```

Comment:
Maybe it's better to use a normal string here, like `Hello @taylor, @"adam driver"`, and not reproduce how this store action works
Because test will pass only if it's aligned to action output => it could differ from real behaviour, and mentions won't work at all